sdk/src/lib.rs

```diff
@@ -35,26 +35,107 @@
-    account_info, address_lookup_table, big_mod_exp, blake3, bpf_loader, bpf_loader_deprecated,
-    bpf_loader_upgradeable, clock, config, custom_heap_default, custom_panic_default,
-    debug_account_data, declare_deprecated_sysvar_id, declare_sysvar_id, ed25519_program,
-    epoch_rewards, epoch_schedule, fee_calculator, impl_sysvar_get, incinerator, instruction,
-    keccak, lamports, loader_instruction, loader_upgradeable_instruction, loader_v4,
-    loader_v4_instruction, message, msg, native_token, nonce, program, program_error,
-    program_option, program_pack, rent, secp256k1_program, serialize_utils, slot_hashes,
-    slot_history, stable_layout, stake, stake_history, syscalls, system_instruction,
-    system_program, sysvar, unchecked_div_by_const, vote,
+    account_info, big_mod_exp, blake3, bpf_loader, bpf_loader_deprecated, clock, config,
+    custom_heap_default, custom_panic_default, debug_account_data, declare_deprecated_sysvar_id,
+    declare_sysvar_id, ed25519_program, epoch_rewards, epoch_schedule, fee_calculator,
+    impl_sysvar_get, incinerator, instruction, keccak, lamports, msg, native_token, program,
+    program_error, program_option, program_pack, rent, secp256k1_program, serialize_utils,
+    slot_hashes, slot_history, stable_layout, syscalls, sysvar, unchecked_div_by_const,
 };
+#[deprecated(
+    since = "2.3.0",
+    note = "Use `solana-address-lookup-table-interface` crate instead"
+)]
+pub mod address_lookup_table {
+    #[allow(deprecated)]
+    pub use solana_program::address_lookup_table::*;
+}
+#[deprecated(
+    since = "2.3.0",
+    note = "Use `solana-loader-v3-interface` crate instead"
+)]
+pub mod bpf_loader_upgradeable {
+    #[allow(deprecated)]
+    pub use solana_program::bpf_loader_upgradeable::*;
+}
+#[deprecated(
+    since = "2.3.0",
+    note = "Use `solana-loader-v2-interface` crate instead"
+)]
+pub mod loader_instruction {
+    #[allow(deprecated)]
+    pub use solana_program::loader_instruction::*;
+}
+#[deprecated(
+    since = "2.3.0",
+    note = "Use solana_loader_v3_interface::instruction instead"
+)]
+pub mod loader_upgradeable_instruction {
+    #[allow(deprecated)]
+    pub use solana_program::loader_upgradeable_instruction::*;
+}
+#[deprecated(since = "2.3.0", note = "Use solana-loader-v4-interface instead")]
+pub mod loader_v4 {
+    #[allow(deprecated)]
+    pub use solana_program::loader_v4::*;
+}
+#[deprecated(
+    since = "2.3.0",
+    note = "Use solana_loader_v4_interface::instruction instead"
+)]
+pub mod loader_v4_instruction {
+    #[allow(deprecated)]
+    pub use solana_program::loader_v4_instruction::*;
+}
+#[deprecated(since = "2.2.0", note = "Use `solana-message` crate instead")]
+pub use solana_message as message;
+#[deprecated(since = "2.3.0", note = "Use `solana-nonce` crate instead")]
+pub mod nonce {
+    #[allow(deprecated)]
+    pub use solana_program::nonce::*;
+}
+#[deprecated(since = "2.3.0", note = "Use `solana-sdk-ids` crate instead")]
```

Comment:
this crate feels like a mistake. I'd rather have individual `solana_x_interface::program` than a junk drawer crate

Reply:
could we not get there with features?

Reply:
i guess i was thinking something like...
```toml
[features]
full = [] # everything but the declare_id!() behind this
default = [full]
```

then if i only want pubkey...
```
[dependencies]
solana_crate = { version = "x.y.z", default-features = false }
```